### PR TITLE
Focus aware query polling

### PR DESCRIPTION
**Pull request by johnnyomair**
Apollo does not support pausing polling when the current browser tab is not in focus (see [feature request](https://github.com/apollographql/apollo-feature-requests/issues/247)). Therefore we add our own `useFocusAwareQuery` hook that wraps `useQuery` and pauses/resumes polling upon focus/blur.

### CODE REVIEW

**Comment by nsams** [comment]
better would be to avoid wrapping the hook, as you can only wrap once - to get one feature - but not twice. Some HOC system that let's the user do comoposition is a possible solution - but overkill in this case. Some kind of additional hook is nice imho (downside: not so easy to use as yours)

possible alternative: comet-admin has it's own `useQuery` with all features we need.

**Review by chernylu — DISMISSED** [review]

**Comment by johnnyomair** [comment]
@nsams what would you say to a solution like this?

```typescript
const { data, startPolling, stopPolling, refetch } = useQuery(...);

useFocusAwarePolling({ pollInterval: 10000, startPolling, stopPolling, refetch });
```

**Comment by nsams** [comment]
:+1:

(not so easy to use, but still preferable for the aforementioned reason)

**Comment by johnnyomair** [comment]
Done in ac427022c37a05790a7aeeaec85e63fa8df888b4. Also added docs on how to use.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ All notable changes to this project will be documented in this file. This projec
 
 -   Add scope support for redirects (`@comet/cms-api` and `@comet/cms-admin`)
 
+### @comet/admin
+
+#### Changes
+
+-   Add `useFocusAwarePolling` hook that can be used in combination with `useQuery` and only fetches when the current browser tab is focused
```
Comment (johnnyomair) [inline CHANGELOG.md:15]:
```suggestion
-   Add `useFocusAwarePolling` hook that can be used in combination with `useQuery` to only fetch when the current browser tab is focused
```

File: packages/admin/admin/src/apollo/useFocusAwarePolling.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { ObservableQuery, OperationVariables, QueryHookOptions } from "@apollo/client";
+import { useEffect } from "react";
+
+type FocusAwarePollingHookOptions<TData = any, TVariables = OperationVariables> = Pick<
+    ObservableQuery<TData, TVariables>,
+    "refetch" | "startPolling" | "stopPolling"
+> &
+    Pick<QueryHookOptions<TData, TVariables>, "pollInterval" | "skip">;
+
+function useFocusAwarePolling<TData = any, TVariables = OperationVariables>({
```
Comment (nsams) [inline packages/admin/admin/src/apollo/useFocusAwarePolling.ts:10]: no need for those generic arguments, when you pick from ObservableQuery the default values are sufficient

**Comment by nsams** [comment]
> Also added docs on how to use.

did you forget to `git add` them?

File: packages/admin/admin/src/apollo/useFocusAwarePolling.ts (hunk 2 above)
Comment (johnnyomair) [reply packages/admin/admin/src/apollo/useFocusAwarePolling.ts]: Done in 47b9cde82b5f4d6fc3e062998051ec4fffccb019.

**Comment by johnnyomair** [comment]
> did you forget to `git add` them?

No, they're right here: bad684adf07a1240fe3cf70fcd257925d653edfe.

**Review by nsams — APPROVED** [review]

**Review by chernylu — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

**Pull request merged**